### PR TITLE
Remove golang gox build dependency

### PR DESCRIPTION
**Pull request by dmboyd**
# Summary

Remove Gox build dependency and enumerated go build targets within Makefile

## Details
Changes are applied to .templates/go/default.mk
Enumerated a set of build targets compatible with golang versions 1.13-1.15
added a minimum supported version warning if someone attempts a cross compilation on an earlier build of golang

## Motivation and Context

This fixes issue #1170 &  #1163 , which cause the build to fail on golang 1.15 or higher. Additionally mitigating additional bugs present within the gox toolchain.
As cross compilation post golang 1.5 is simplified to a command line option, the use of a separate dependency isn't neccesary.

## How Has This Been Tested?

Tested locally - golang 1.15 Darwin/amd64 -- no golang build failures
tested with:
`make build with make docker-run (golang 1.13)`
no go build failures

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).

## Checklist:

- [ ] The change has been ported to Java.
- [ ] The change has been ported to Ruby.
- [ ] The change has been ported to JavaScript.
- [x] The change has been ported to Go.
- [ ] The change has been ported to .NET.
- [ ] I've added tests for my code.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the CHANGELOG accordingly.

### CODE REVIEW

**Comment by aslakhellesoy** [comment]
Thanks for this! I'll try to review it this week.

File: .templates/go/default.mk [hunk 1]
```diff
@@ -30,23 +42,43 @@ endif
...
+$(EXE): .deps $(GO_SOURCE_FILES)
 	# Compile executable for the local platform only
 	go build -ldflags $(GOX_LDFLAGS) -o $@ ./cmd
 endif
 
+$(PLATFORMS): .deps $(GO_SOURCE_FILES) supported-go-version
+	mkdir -p dist
+	# Cross-compile executable for many platforms
+	echo "Building $(X-OS)-$(X-ARCH)"
+	GOOS=$(X-OS) GOARCH=$(X-ARCH) go build -buildmode=exe -ldflags $(GOX_LDFLAGS) -o "dist/$(EXE_BASE_NAME)-$(X-OS)-$(X-ARCH)" -a ./cmd
```
Comment (aslakhellesoy) [inline .templates/go/default.mk:67]: Let's rename `GOX_LDFLAGS` to `LDFLAGS` now that we're no longer using gox.

**Review by aslakhellesoy — CHANGES_REQUESTED** [review]
Lovely to see proper Makefile fu! If you could please address my tiny nitpick I'll merge this.

**Review by aslakhellesoy — CHANGES_REQUESTED** [review]
Also, please remove all references to `gox` in all `go.mod` files.

File: .templates/go/default.mk (hunk 1 above)
Comment (dmboyd) [reply .templates/go/default.mk]: Adjusted terminology to adjust LDFLAG name convention

**Review by aslakhellesoy — APPROVED** [review]
Excellent - thanks a lot!

**Comment by aslakhellesoy** [comment]
Hi @dmboyd,

Thanks for your making your first contribution to Cucumber, and welcome to the Cucumber committers team! You can now push directly to this repo and all other repos under the cucumber organization! 🍾

In return for this generous offer we hope you will:

* ✅ Continue to use branches and pull requests. When someone on the core team approves a pull request (yours or someone else's), you're welcome to merge it yourself.
* 💚 Commit to setting a good example by following and upholding our [code of conduct](https://cucumber.io/conduct) in your interactions with other collaborators and users.
* 💬 Join the [community Slack channel](http://cucumber.io/support#slack) to meet the rest of the team and make yourself at home.
* ℹ️ Don't feel obliged to help, just do what you can if you have the time and the energy.
* 🙋 Ask if you need anything. We're looking for feedback about how to make the project more welcoming, so please tell us!

On behalf of the Cucumber core team,
Aslak Hellesøy
Creator of Cucumber

**Pull request merged**